src/services/FirebaseService.ts

```diff
@@ -0,0 +1,58 @@
+import { notEqual } from 'assert';
+import admin from 'firebase-admin';
+import { inject, injectable } from 'inversify';
+import { IApiFactory } from '../client/ApiFactory';
+import { DappItem } from '../models/Dapp';
+import { NetworkType } from '../networks';
+
+export interface IFirebaseService {
+    getDapps(network: NetworkType): Promise<DappItem[]>;
+}
+
+@injectable()
+export class FirebaseService implements IFirebaseService {
+    private app: admin.app.App;
+
+    constructor(@inject('factory') private _apiFactory: IApiFactory) {}
+
+    public async getDapps(network: NetworkType = 'astar'): Promise<DappItem[]> {
+        this.initApp();
+
+        const collectionKey = await this.getCollectionKey(network);
+        const query = admin.firestore().collection(collectionKey);
+        const data = await query.orderBy('name').get();
+
+        const result: DappItem[] = [];
+        data.forEach((x) => {
+            const data = x.data() as DappItem;
+            result.push(data);
+        });
+
+        return result;
+    }
+
+    private initApp(): void {
+        if (!this.app) {
+            this.app = admin.initializeApp({
+                credential: admin.credential.cert({
+                    projectId: process.env.FIREBASE_PROJECT_ID,
```

Comment:
Can you add the environment variables on Heroku?

Reply:
Sure. Variables have been addeed